### DISCUSSION TITLE
Inverse of torch_geometric.transforms.NormalizeFeatures()

### DISCUSSION THREAD

**Discussion by CodingMaker2** [pyg-team/pytorch_geometric#6996] [Q&A] [answered]
How do I restore the converted data of the NormalizeFeatures() function to the original data?

**Comment by rusty1s** [comment, marked as answer]
This is not possible without storing more information, as you cannot go from sum->mean and then back from mean->sum without knowing the factor you how to multiply with.

**Reply by CodingMaker2** [reply]
for store in data.stores:
       for key, value in store.items(*self.attrs):
              value = value - value.min()
              value.div_(value.sum(dim=-1, keepdim=True).clamp_(min=1.))
              store[key] = value
return data

I have noticed that there is a normalization process in the source code. Is it possible to use the 'value. sum (dim=- 1, keepdim=True)' in this code as the factor to multiple with?

**Reply by rusty1s** [reply]
Yes, that's what you would need to do in this case.

**Reply by CodingMaker2** [reply]
Thanks for your answer, I will have a try.
